xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecIMX.cpp

```diff
@@ -1507,61 +1535,52 @@ bool CIMXContext::SetVSync(bool enable)
   return true;
 }
 
-void CIMXContext::SetDoubleRate(bool flag)
+void CIMXContext::SetBlitRects(const CRect &srcRect, const CRect &dstRect)
 {
-  if (flag)
-    m_currentFieldFmt |= IPU_DEINTERLACE_RATE_EN;
-  else
-    m_currentFieldFmt &= ~IPU_DEINTERLACE_RATE_EN;
-
-  m_currentFieldFmt &= ~IPU_DEINTERLACE_RATE_FRAME1;
+  m_srcRect = srcRect;
+  m_dstRect = dstRect;
 }
 
-bool CIMXContext::DoubleRate() const
+inline
+void CIMXContext::SetFieldData(uint8_t fieldFmt)
 {
-  return m_currentFieldFmt & IPU_DEINTERLACE_RATE_EN;
-}
+  if (m_bStop || !IsRunning())
+    return;
 
-void CIMXContext::SetInterpolatedFrame(bool flag)
-{
-  if (flag)
-    m_currentFieldFmt &= ~IPU_DEINTERLACE_RATE_FRAME1;
-  else
-    m_currentFieldFmt |= IPU_DEINTERLACE_RATE_FRAME1;
-}
+  // disable deinterlacing when not rendering fullscreen
+  if (!g_graphicsContext.IsFullScreenVideo())
```

Comment:
decoder should not care about this. player sets flag to control decoder:
https://github.com/xbmc/xbmc/blob/32af4975182bdb9e07559862731810067c020bf0/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodec.h#L148